spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/OAuth2AuthenticationConverter.java

```diff
@@ -0,0 +1,36 @@
+package com.sap.cloud.security.xsuaa.token;
+
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.provider.AuthorizationRequest;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+
+import com.sap.cloud.security.xsuaa.extractor.AuthoritiesExtractor;
+
+/**
+ * An authentication converter that sets a OAuth2Authentication object.
+ * Required to keep compatibility with UAA.
+ */
+public class OAuth2AuthenticationConverter extends AbstractAuthenticationConverter {
+
+
+	public OAuth2AuthenticationConverter(AuthoritiesExtractor authoritiesExtractor) {
+		super(authoritiesExtractor);
+	}
+
+	@Override
+	public OAuth2Authentication convert(Jwt jwt) {
```

Comment:
if this subclasses TokenAuthenticationConverter, then it can do the following:

```java
@Override
public OAuth2Authentication convert(Jwt jwt) {
      AuthenticationToken authenticationToken = super.convert(jwt);
      String clientId = jwt.getClaimAsString(CLAIM_CLIENT_ID);
      Collection<String> scopes = authenticationToken.getAuthorities().stream()
				.map(object -> Objects.toString(object, null))
				.collect(Collectors.toList());
      AuthorizationRequest authorizationRequest = new AuthorizationRequest(clientId, scopes);
      authorizationRequest.setApproved(true);
      authorizationRequest.setAuthorities(authenticationToken.getAuthorities()); // pls check if constructor AND method call are required

     return new OAuth2Authentication(authorizationRequest.createOAuth2Request(), authenticationToken);
}

```